src/behavior-considered-undefined.md

```diff
@@ -98,6 +101,24 @@ reading uninitialized memory is permitted are inside `union`s and in "padding"
 
 The span of bytes a pointer or reference "points to" is determined by the pointer value and the size of the pointee type (using `size_of_val`).
 
+### Places based on misaligned pointers
+[based on a misaligned pointer]: #places-based-on-misaligned-pointers
+
+A place is said to be "based on a misaligned pointer" if the last `*` projection
+during place computation was performed on a pointer that was not aligned for its
+type. (If there is no `*` projection in the place expression, then this is
+accessing the field of a local and rustc will guarantee proper alignment. If
+there are multiple `*` projection, then each of them incurs a load of the
+pointer-to-be-dereferenced itself from memory, and each of these loads is
+subject to the alignment constraint.)
```

Comment:
It might be worthwhile to add a note here that e.g.

> `*` projections can be implicitly introduced even without a literal source `*` when references are involved.

Reply:
References are always aligned though so this cannot lead to UB.

Reply:
So if you have `(*misaligned).tref.field`, that's actually `(*(*misaligned).tref).field` after desugar due to reference autoderef, so it's the multiple `*` projections case and the misaligned pointer is loaded from, causing UB. It's not surprising after you consider the desugar, but I think it's relevant to note here since the condition of only being a single `*` projection is not purely syntactical in the surface language when references may be involved.

Reply:
`(*misaligned).tref.field` looks like it requires `misaligned` to be aligned. `(*(*misaligned).tref).field` looks like it requires `misaligned` and `tref` to be aligned, but `tref` is a reference so there is already very explicit documentation saying it has to always be aligned.

I don't see how there is a chance of missing UB here.

Reply:
Are you maybe referring to a different part of the diff, where it says "Accessing (loading from or storing to) a place ..."? Indeed I haven't defined this entirely in terms of surface syntax, I am relying on other parts of the documentation saying when place accesses happen.